source/class/cv/io/timeseries/Plugin.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2024, Christian Mayer and the CometVisu contributors.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
+ *
+ */
+
+/**
+ * Base class for plugins that can be used as time series sources.
+ * @since 2024
+ * @author Tobias Bräutigam
+ */
+qx.Class.define('cv.io.timeseries.Plugin', {
+  extend: cv.io.timeseries.AbstractTimeSeriesSource,
+
+  /*
+  ***********************************************
+    STATICS
+  ***********************************************
+  */
+  statics: {
+    _registry: {},
+
+    registerPlugin(type, clazz) {
+      this._registry[type] = clazz;
+    }
+  },
+
+  /*
+  ***********************************************
+    MEMBERS
+  ***********************************************
+  */
+  members: {
+    _plugin: null,
+
+    _init() {
+      const config = this.getConfig();
+      if (!Object.prototype.hasOwnProperty.call(cv.io.timeseries.Plugin._registry, config.subType)) {
+        throw new Error('Unknown plugin type: ' + config.subType);
+      }
+      this._plugin = new cv.io.timeseries.Plugin._registry[config.subType](config, this._chart);
+    },
+
+    /**
+     * Options to configure the builtin fetch function.
+     *
+     * @typedef {Object} FetchRequestConfig
+     * @property {url} string - The URL to fetch the data from
```

Comment:
It should be `url {string}`, `options {Object}`, `proxy {boolean}`, ... and not the other way round. That's for all lines in this file.

Reply:
I went for the official jsdoc-way of ordering that. Qx want to switch to that way (its on the roadmap including some tool that can alter existing code on-the-fly). In the qx-doc you can find both way, so I thought they already support the jsdoc-ordering and saw no reason to use the legacy way any further.
But I will check if the apiviewer handles this correctly first. If not I will change it back.

Reply:
Qx supports both ways, we already use both ways, so lets stick to the official jsdoc way from now on. And wait for a migration script mentioned on qx-roadmap to get it consistent.